crates/ruff_python_literal/src/format.rs

```diff
@@ -1236,6 +1406,22 @@ mod tests {
         assert_eq!(FormatString::from_str("abcd{1}:{key}"), expected);
     }
 
+    #[test]
+    fn test_format_parse_nested_replacement() {
+        let expected = Ok(FormatString {
+            format_parts: vec![
+                FormatPart::Literal("abcd".to_owned()),
+                FormatPart::Field {
+                    field_name: "1".to_owned(),
+                    conversion_spec: None,
+                    format_spec: "{a}".to_owned(),
+                },
+            ],
+        });
+
+        assert_eq!(FormatString::from_str("abcd{1:{a}}"), expected);
```

Comment:
Not for this PR, but I feel like these are good candidates for `insta::assert_debug_snapshot`, so that we don't have to write out and maintain the expectations manually.

Reply:
Makes sense, I tried to match the existing pattern but I agree that's more maintainable.